ReSharper.FSharp/src/FSharp.Psi.Intentions/src/QuickFixes/ReplaceWithTripleQuotedInterpolatedStringFix.fs

```diff
@@ -0,0 +1,96 @@
+open JetBrains.ReSharper.Plugins.FSharp.Psi.Parsing
+open JetBrains.ReSharper.Plugins.FSharp.Psi.Services.Util
+open JetBrains.ReSharper.Plugins.FSharp.Psi.Tree
+open JetBrains.ReSharper.Psi.ExtensionsAPI.Tree
+open JetBrains.ReSharper.Psi.Parsing
+open JetBrains.ReSharper.Psi.Tree
+open JetBrains.ReSharper.Plugins.FSharp.Psi.Features.StringLiteralsUtil
+open JetBrains.ReSharper.Resources.Shell
+open JetBrains.Text
+
+type ReplaceWithTripleQuotedInterpolatedStringFix(error: SingleQuoteInSingleQuoteError) =
+    inherit FSharpQuickFixBase()
+
+    let createStart content =
+        FSharpTokenType.TRIPLE_QUOTE_INTERPOLATED_STRING_START.Create($"$\"\"\"{content}{{")
+
+    let createMiddle content =
+        FSharpTokenType.TRIPLE_QUOTE_INTERPOLATED_STRING_MIDDLE.Create($"}}{content}{{")
+
+    let createEnd content =
+        FSharpTokenType.TRIPLE_QUOTE_INTERPOLATED_STRING_END.Create($"}}{content}\"\"\"")
+
+    let checkRegularStringLiteral (literal: ITokenNode) =
+        let tokenContent = getStringContent (literal.GetTokenType()) (literal.GetText())
+        let lexer = RegularInterpolatedStringLexer(StringBuffer(tokenContent))
+
+        let mutable found = false
+        while not found && lexer.CanAdvance do
+            lexer.Advance()
+            found <- lexer.TokenType == StringTokenTypes.ESCAPE_CHARACTER
+        found
+
+    let processRegularStringLiteral (literal: ITokenNode)  =
+        let text = literal.GetText()
+        let tokenType = literal.GetTokenType()
+        let content = getStringContent tokenType text
+
+        let resultingNode: ITreeNode =
+            match tokenType with
+            | tokenType when tokenType == FSharpTokenType.REGULAR_INTERPOLATED_STRING_START -> createStart content
+            | tokenType when tokenType == FSharpTokenType.REGULAR_INTERPOLATED_STRING_MIDDLE -> createMiddle content
+            | tokenType when tokenType == FSharpTokenType.REGULAR_INTERPOLATED_STRING_END -> createEnd content
+            | _ -> literal
+
+        if literal != resultingNode then
+            ModificationUtil.ReplaceChild(literal, resultingNode) |> ignore
+
+    let processVerbatimStringLiteral (literal: ITokenNode) =
+        let text = literal.GetText()
+        let tokenType = literal.GetTokenType()
+        let content = (getStringContent tokenType text).Replace("\"\"", "\"")
+
+        let resultingNode: ITreeNode =
+            match tokenType with
+            | tokenType when tokenType == FSharpTokenType.VERBATIM_INTERPOLATED_STRING_START -> createStart content
+            | tokenType when tokenType == FSharpTokenType.VERBATIM_INTERPOLATED_STRING_MIDDLE -> createMiddle content
+            | tokenType when tokenType == FSharpTokenType.VERBATIM_INTERPOLATED_STRING_END -> createEnd content
+            | _ -> literal
+
+        if literal != resultingNode then
+            ModificationUtil.ReplaceChild(literal, resultingNode) |> ignore
```

Comment:
Could you remove the duplication here? The only meaningful difference is quotes replacement, which could be controlled by a param.